File: 02_ProgrammingFundamentals/_09_MapsLambdaAndStreamAPI/_02_Exercise/_08_CompanyUsers.java
package _02_ProgrammingFundamentals._09_MapsLambdaAndStreamAPI._02_Exercise;

import java.util.*;
import java.util.stream.Collectors;

public class _08_CompanyUsers {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input value
        String input = scan.nextLine();

        // Add variable
        Map<String, Set<String>> employees = new LinkedHashMap<>();

        // Make employee database
        while (!input.equals("End")) {
            String[] line = input.split(" -> ");

            String company = line[0];
            String id = line[1];

            if (!employees.containsKey(company)) {
                employees.put(company, new LinkedHashSet<>());
            }

            employees.get(company).add(id);

            input = scan.nextLine();
        }

        // Print result
        employees.entrySet().forEach(element -> {
            String company = element.getKey();
            String ids = element.getValue().stream().map(string -> String.format("-- %s", string)).collect(Collectors.joining("\n"));

            System.out.printf("%s%n%s%n", company, ids);
        });

        scan.close();

//        employees.entrySet().forEach(element -> {
//            String company = element.getKey();
//            String ids = element.getValue().stream().sorted().map(string -> String.format("-- %s", string)).collect(Collectors.joining("\n"));
//
//            System.out.printf("%s%n%s%n", company, ids);
//        });


//        employees.entrySet().forEach(element -> {
//            String company = element.getKey();
//            String ids = element.getValue().stream().sorted(String::compareTo).map(string -> String.format("-- %s", string)).collect(Collectors.joining("\n"));
//
//            System.out.printf("%s%n%s%n", company, ids);
//        });
    }
}

File: 24_C#Fundamentals/_07_AssociativeArrays/_02_Exercise/_08_CompanyUsers/_08_CompanyUsers.cs
using System;
using System.Collections.Generic;
using System.Xml.Linq;

public class _08_CompanyUsers
{
    public static void Main(string[] args)
    {
        // Read input value
        string input = Console.ReadLine();

        // Add variable
        Dictionary<string, HashSet<string>> employees = new Dictionary<string, HashSet<string>>();

        // Make employee database
        while (!input.Equals("End"))
        {
            string[] line = input.Split(" -> ");

            string company = line[0];
            string id = line[1];

            if (!employees.ContainsKey(company))
            {
                employees[company] = new HashSet<string>();
            }

            employees[company].Add(id);

            input = Console.ReadLine();
        }

        // Print result
        foreach (var kvp in employees)
        {
            Console.WriteLine(kvp.Key);
            foreach (var id in kvp.Value)
            {
                Console.WriteLine($"-- {id}");
            }
        }
    }
}
